direct/data/datasets.py

```diff
@@ -395,6 +400,259 @@ def __broadcast_mask(self, kspace_shape, mask):
         return mask
 
 
+class CMRxReconDataset(Dataset):
+    """CMRxRecon Challenge Dataset [1]_.
+
+    References
+    ----------
```

Comment:
My impression is that all the datasets up to now should be able to be subclasses of H5SlideDataset or something like that. Can you look into if you can de-duplicate code that way?

Reply:
This is not an `h5` dataset. It's matlab files (`.mat`). I tried to minimise d

Reply:
Would it be possible to define a class function "read_region" or some sort, and then define that as an abstract base class?

Reply:
I don't understand but I think the way this is is fine. The dublicate code is minimum. 

And since every dataset is different, its unlikely that inheriting from H5SliceDataset or such classes is needed

Reply:
Of course I wouldn't mind if you wanna try this out, but I think it's not needed.